.github/workflows/ci-kafka.yml

```diff
@@ -27,7 +27,17 @@ jobs:
         go-version: 1.20.x
 
     - name: Run kafka integration tests
-      run: bash scripts/kafka-integration-test.sh
+      run: |
+            retrycount=0
+             maxRetries=3
+             backoff=1
+             while[$retrycount -lt $maxRetries];do
+              bash scripts/kafka-integration-test.sh && break
+             echo "Tests failed, retrying in $backoff seconds..."
+             sleep $backoff
+            retryCount=$((retryCount+1))
+            backoff=$((backoff*2))
+            done
```

Comment:
What do folks think about placing this logic within the `scripts/kafka-integration-test.sh` itself? It's not used anywhere else, and is easier to test locally.

Reply:
I am not yet convinced it's even necessary. Generally CITs fail because the backend storage is not ready, so it's usually better to have a more reliable health check for storage than to repeat the test itself. In case of Kafka specifically, the tests started failing not just because it was "not ready", but because it was not getting ready even with retries, likely because of some configuration changes required for the new Kafka image version.